src/cf.data.pre

```diff
@@ -4212,8 +4212,40 @@ DOC_START
 		[http::]rm	Request method (GET/POST etc)
 		[http::]>rm	Request method from client
 		[http::]<rm	Request method sent to server or peer
-		[http::]ru	Request URL from client (historic, filtered for logging)
-		[http::]>ru	Request URL from client
+
+		[http::]ru	Request URL received (or computed) and sanitized
+
+				Logs request URI received from the client, a
+				request adaptation service, or a request
+				redirector (whichever was applied last). For
+				historical reasons, whitespace in this field
+				is sanitized according to uri_whitespace. Also,
+				the URI query string is stripped out according to
+				strip_query_terms. That cleanup happens before
+				field encoding is applied, leading to confusing
+				double-encoding results in some non-default
```

Comment:
After revising the code I think that 'double-encoding' problem is hardly possible for %>ru and even not possible for %ru because all url-path related methods (urlParse(), setLogUri() and urlCanonicalClean()) use RFC1738_ESCAPE_UNESCAPED flag in rfc1738_* methods. The only place of 'double encoding' I can think of is the encoding applied in Format.cc (for %>ru) where rfc1738_escape() is called without such flag. However, this happens only with a specific condition (fmt->quote==LOG_QUOTE_URL).

Reply:
> The only place of 'double encoding' I can think of is the encoding applied in Format.cc

Yes, double-encoding I documented (and tested) required Format.cc participation IIRC.

> The only place of 'double encoding' I can think of is the encoding applied in Format.cc (**for `%>ru`**)

Why not for `%ru` as well? Would not `%#ru` suffer from the same double-encoding problem as %#>ru` would?

> However, this happens only with a specific condition (fmt->quote==LOG_QUOTE_URL).

... which is one of the reasonable settings for %ru and %>ru -- the admin knows they are logging a URL so they are not that unlikely to specify URL encoding for it.

AFAICT, this double-encoding cannot happen by default (because default `fmt->quote` is `LOG_QUOTE_NONE` which skips `%`) so my implication that "%>ru is encoded by default" and, hence, would suffer from double-encoding in more cases was incorrect. However, AFAICT, double-encoding is very much possible with explicitly configured URL encoding (`%#...`) for both `%ru` and `%>ru`, right?

Reply:
> double-encoding is very much possible with explicitly configured URL encoding (%#...) for both %ru and %>ru, right?

Agreed, I missed the second condition "fmt->quote != LOG_QUOTE_NONE" in Format.cc, making possible double-encoding for %#ru.

Reply:
I pushed documentation fixes into the branch (27f085e). Please re-check the cumulative cf.data.pre changes in this branch and let me know if they still have bugs (that you cannot fix).